frontend/templates/product-list/sections/FilterableProductsSection/index.tsx

```diff
@@ -65,6 +66,31 @@ export function FilterableProductsSection({
       () => filterFalsyItems(hits.map(productPreviewFromAlgoliaHit)),
       [hits]
    );
+
+   const [prevProductIds, setPrevProductIds] = React.useState<string[]>([]);
+
+   React.useEffect(() => {
+      const currentProductIds = products.map((product) => product.id);
+      if (
+         JSON.stringify(currentProductIds) !== JSON.stringify(prevProductIds)
+      ) {
+         if (currentProductIds.length > 0) {
+            debouncedTrackGA4ViewItemList({
+               item_list_id: productList.handle,
+               item_list_name: productList.title,
+               items: products.map((product) => ({
+                  item_id: product.sku,
+                  item_name: product.title,
+                  item_variant: product.id,
+                  quantity: product.quantityAvailable,
+                  price: product.price.amount,
+               })),
+            });
+         }
+      }
+      setPrevProductIds(currentProductIds);
+   }, [products]);
```

Comment:
Warning

Reply:
Products is already in the dependencies so I'd already this to not trigger on a duplicate view. Not sure why we need the JSON stringify thing. It's possible this double-triggered because of react strict mode in dev.

Reply:
its somesort of quick rerender or something with the `hits` i believe. didn't figure it out, but as far as i can tell we want to make sure we don't log everytime the filterableProductsSection rerenders